stock_quant_package_product_packaging/models/stock_quant_package.py

```diff
@@ -43,12 +56,42 @@ def auto_assign_packaging(self):
                 and pack.single_product_id
                 and pack.single_product_qty
             ):
-                packaging = self.env["product.packaging"].search(
-                    [
-                        ("product_id", "=", pack.single_product_id.id),
-                        ("qty", "=", pack.single_product_qty),
-                    ],
-                    limit=1,
+                pack._assign_package_type(
+                    pack.single_product_id, pack.single_product_qty
                 )
-                if packaging:
-                    pack.write({"product_packaging_id": packaging.id})
+
+    def _assign_package_type(self, product, quantity):
+        self.ensure_one()
+        packaging = product._find_best_packaging(quantity)
+        if packaging and packaging.qty == quantity:
+            self.write({"product_packaging_id": packaging.id})
+        elif packaging:
+            self._sync_package_type_from_packaging(packaging)
+        else:
+            self._sync_package_type_from_single_product()
+
+        if not self.package_type_id and product.package_type_id:
+            self.package_type_id = product.package_type_id
+
+    def _sync_package_type_from_packaging(self, packaging):
+        for package in self:
+            if package.package_type_id:
+                # Do not set package type for delivery packages
+                # to not trigger constraint like height requirement
+                # (we are delivering them, not storing them)
+                continue
+            package_type = packaging.package_type_id
+            if not package_type:
+                continue
+            package.package_type_id = package_type
+
+    def _sync_package_type_from_single_product(self):
+        for package in self:
+            if package.single_product_id and package.package_type_id:
```

Comment:
there is a product by definition
```suggestion
            if package.package_type_id:
```